optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/persistence/MeetingSchedulingXlsxFileIO.java

```diff
@@ -0,0 +1,722 @@
+                                                  filteredConstraintNames);
+                    mergePreviousTimeGrain = meetingAssignmentList.isEmpty() ? null : timeGrain;
+                    columnMergeStart = currentColumnNumber;
+                    int longestDurationInGrains = 1;
+                    for (MeetingAssignment meetingAssignment : meetingAssignmentList) {
+                        if (meetingAssignment.getMeeting().getDurationInGrains() > longestDurationInGrains) {
+                            longestDurationInGrains = meetingAssignment.getMeeting().getDurationInGrains();
+                        }
+                    }
+                    previousMeetingRemainingTimeGrains = longestDurationInGrains - 1;
+                }
+            }
+        }
+
+        private void writeTimeGrainDaysHeaders() {
+            Day previousTimeGrainDay = null;
+            int mergeStart = -1;
+
+            for (TimeGrain timeGrain : solution.getTimeGrainList()) {
+                Day timeGrainDay = timeGrain.getDay();
+                if (timeGrainDay.equals(previousTimeGrainDay)) {
+                    nextHeaderCell("");
+                } else {
+                    if (previousTimeGrainDay != null) {
+                        currentSheet.addMergedRegion(new CellRangeAddress(currentRowNumber, currentRowNumber, mergeStart, currentColumnNumber));
+                    }
+                    nextHeaderCell(DAY_FORMATTER.format(
+                        LocalDate.ofYearDay(Year.now().getValue(), timeGrainDay.getDayOfYear())));
+                    previousTimeGrainDay = timeGrainDay;
+                    mergeStart = currentColumnNumber;
+                }
+            }
+            if (previousTimeGrainDay != null) {
+                currentSheet.addMergedRegion(new CellRangeAddress(currentRowNumber, currentRowNumber, mergeStart, currentColumnNumber));
+            }
+        }
+
+        private void writeTimeGrainHoursHeaders() {
+            for (TimeGrain timeGrain : solution.getTimeGrainList()) {
+                LocalTime startTime = LocalTime.ofSecondOfDay(timeGrain.getStartingMinuteOfDay() * 60);
+                LocalTime endTime = LocalTime.ofSecondOfDay(
+                    (timeGrain.getStartingMinuteOfDay() + timeGrain.GRAIN_LENGTH_IN_MINUTES) * 60);
+                nextHeaderCell(TIME_FORMATTER.format(startTime));
+            }
+        }
+
+        protected void nextMeetingAssignmentListCell(List<MeetingAssignment> meetingAssignmentList, Function<MeetingAssignment, String> stringFunction,
+                                                     String[] filteredConstraintNames) {
+            nextMeetingAssignmentListCell(false, meetingAssignmentList, stringFunction, filteredConstraintNames);
+        }
+
+        protected void nextMeetingAssignmentListCell(boolean unavailable, List<MeetingAssignment> meetingAssignmentList,
```

Comment:
The same suggestion as above - could you please structure this code into smaller methods?